addons/hr_timesheet/controllers/portal.py

```diff
@@ -135,17 +135,24 @@ def get_timesheets():
                     raw_timesheets_group = Timesheet_sudo._read_group(
                         domain, ['date:day'], ['unit_amount:sum', 'id:recordset'], order='date:day desc'
                     )
-                    grouped_timesheets = [(records, unit_amount) for __, unit_amount, records in raw_timesheets_group]
+                    grouped_timesheets = [(records, unit_amount, 0) for __, unit_amount, records in raw_timesheets_group]
 
                 else:
                     time_data = Timesheet_sudo._read_group(domain, [field], ['unit_amount:sum'])
-                    mapped_time = {field.id: unit_amount for field, unit_amount in time_data}
-                    grouped_timesheets = [(Timesheet_sudo.concat(*g), mapped_time[k.id]) for k, g in groupbyelem(timesheets, itemgetter(field))]
+                    mapped_time = {
+                        field.id: (unit_amount, field.allocated_hours if groupby in ['project_id', 'task_id', 'parent_task_id'] else 0)
```

Comment:
```suggestion
                        project_related_group_by = groupby in ['project_id', 'task_id', 'parent_task_id']
                        ....
                        field.id: (unit_amount, project_related_group_by and field.allow_timesheets, field.allocated_hours if project_related_group_by else 0)
```

Reply:
this comment was not applied, my tuple contains 3 items (unit_amount, bool, allocated_hours) but yours contains only 2 fields, the bool is not here, for some cases, the bool is here with value False and for this case, you called it should_display (bool name) and then you used it as float. 

you also missed my last comment 
![image](https://github.com/user-attachments/assets/50437088-bda8-4a89-9308-49651733b88d)

you deleted allocated_hours (as you don't have it in the tuple) so we use the bool to decided (display or not) then we display the allocated_hours, let me know if my suggestion is not clear